src/test/java/seedu/address/model/module/UniqueModuleListTest.java

```diff
@@ -0,0 +1,167 @@
+package seedu.address.model.module;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalModules.MODULE_1;
+import static seedu.address.testutil.TypicalModules.MODULE_2;
+import static seedu.address.testutil.TypicalModules.MODULE_NAME_0;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.module.exceptions.DuplicateModuleException;
+import seedu.address.model.module.exceptions.ModuleNotFoundException;
+import seedu.address.testutil.ModuleBuilder;
+
+class UniqueModuleListTest {
+    private final UniqueModuleList uniqueModuleList = new UniqueModuleList();
+
+    @Test
+    public void contains_nullModule_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueModuleList.contains(null));
+    }
+
+    @Test
+    public void contains_moduleNotInList_returnsFalse() {
+        assertFalse(uniqueModuleList.contains(MODULE_1));
+    }
+
+    @Test
+    public void contains_moduleInList_returnsTrue() {
+        uniqueModuleList.add(MODULE_1);
+        assertTrue(uniqueModuleList.contains(MODULE_1));
+    }
+
+    @Test
+    public void contains_moduleWithSameIdentityFieldsInList_returnsTrue() {
+        uniqueModuleList.add(MODULE_1);
+        Module editedModule1 = new ModuleBuilder().withName(MODULE_NAME_0).build();
+        assertTrue(uniqueModuleList.contains(editedModule1));
+    }
+
+    @Test
+    public void add_nullModule_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueModuleList.add(null));
+    }
+
+    @Test
+    public void add_duplicateModule_throwsDuplicateModuleException() {
+        uniqueModuleList.add(MODULE_1);
+        assertThrows(DuplicateModuleException.class, () -> uniqueModuleList.add(MODULE_1));
+    }
+
+    @Test
+    public void setModule_nullTargetModule_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueModuleList.setModule(null, MODULE_1));
+    }
+
+    @Test
+    public void setModule_nullEditedModule_throwsNullPointerException() {
+        uniqueModuleList.add(MODULE_1);
+        assertThrows(NullPointerException.class, () -> uniqueModuleList.setModule(MODULE_1, null));
+    }
+
+    @Test
+    public void setModule_targetModuleNotInList_throwsModuleNotFoundException() {
+        assertThrows(ModuleNotFoundException.class, () -> uniqueModuleList.setModule(MODULE_1, MODULE_1));
+    }
+
+    @Test
+    public void setModule_editedModuleIsSameModule_success() {
```

Comment:
Would the edited module being the same module result in a command exception instead?

https://github.com/AY2122S1-CS2103-F09-1/tp/pull/211/commits/839d17a87f955c5683e15d14536d9f6817102b3f

Reply:
This is the test for the method `setModule` under the class `UniqueModuleList`, not for the class `EditModuleCommand`. This method does not throw any exception.